pinot-common/src/main/java/org/apache/pinot/common/request/GroupBy.java

```diff
@@ -550,13 +547,13 @@ public void read(org.apache.thrift.protocol.TProtocol iprot, GroupBy struct) thr
           case 3: // EXPRESSIONS
             if (schemeField.type == org.apache.thrift.protocol.TType.LIST) {
               {
-                org.apache.thrift.protocol.TList _list65 = iprot.readListBegin();
-                struct.expressions = new java.util.ArrayList<java.lang.String>(_list65.size);
-                @org.apache.thrift.annotation.Nullable java.lang.String _elem66;
-                for (int _i67 = 0; _i67 < _list65.size; ++_i67)
+                org.apache.thrift.protocol.TList _list73 = iprot.readListBegin();
+                struct.expressions = new java.util.ArrayList<java.lang.String>(_list73.size);
+                @org.apache.thrift.annotation.Nullable java.lang.String _elem74;
+                for (int _i75 = 0; _i75 < _list73.size; ++_i75)
                 {
-                  _elem66 = iprot.readString();
-                  struct.expressions.add(_elem66);
+                  _elem74 = iprot.readString();
+                  struct.expressions.add(_elem74);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 549:559, 523:533.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.